cmd/minishift/cmd/oc_env.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+
+	"github.com/docker/machine/libmachine"
+	instanceState "github.com/minishift/minishift/pkg/minishift/config"
+	utilCmd "github.com/minishift/minishift/pkg/util/cmd"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+	"os"
+	"text/template"
+)
+
+const (
+	ocEnvTmpl = `{{ .Prefix }}PATH{{ .Delimiter }}{{ .OcPath }}{{ .Suffix }}`
+)
+
+type OcShellConfig struct {
+	utilCmd.ShellConfig
+	OcPath    string
+	UsageHint string
+}
+
+func getOcConfigSet(api libmachine.API, forceShell string) (*OcShellConfig, error) {
+	userShell, err := utilCmd.GetShell(forceShell)
+	if err != nil {
+		return nil, err
+	}
+
+	cmdLine := "minishift oc-env"
+	shellCfg := &OcShellConfig{
+		OcPath:    instanceState.InstanceConfig.OcPath,
+		UsageHint: utilCmd.GenerateUsageHint(userShell, cmdLine),
+	}
+
+	prefix, suffix, delimiter := utilCmd.GetPrefixSuffixDelimiterForSet(userShell, true)
+	shellCfg.Prefix = prefix
+	shellCfg.Suffix = suffix
+	shellCfg.Delimiter = delimiter
+
+	return shellCfg, nil
+}
+
+// ************ Seems to be producing "unset PATH" which will unset whole PATH ******************
+
+//func getOcConfigUnset(api libmachine.API, forceShell string) (*OcShellConfig, error) {
+//	userShell, err := utilCmd.GetShell(forceShell)
+//	if err != nil {
+//		return nil, err
+//	}
+//
+//	cmdLine := "minishift oc-env"
+//	shellCfg := &OcShellConfig{
+//		UsageHint: utilCmd.GenerateUsageHint(userShell, cmdLine),
+//	}
+//
+//	prefix, suffix, delimiter := utilCmd.GetPrefixSuffixDelimiterForUnSet(userShell)
+//	shellCfg.Prefix = prefix
+//	shellCfg.Suffix = suffix
+//	shellCfg.Delimiter = delimiter
+//
+//	return shellCfg, nil
+//}
+
+func executeOcTemplateStdout(shellCfg *OcShellConfig) error {
+	tmpl := template.Must(template.New("envConfig").Parse(ocEnvTmpl))
+	return tmpl.Execute(os.Stdout, shellCfg)
+}
+
+var ocEnvCmd = &cobra.Command{
+	Use:   "oc-env",
+	Short: "Sets path of 'oc' binary.",
+	Long:  `Sets path of OpenShift client binary, 'oc'.`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+		defer api.Close()
+
+		fmt.Println("oc-env command")
```

Comment:
Here we need to add more information. Similar to _docker-env_. Something like:

   export PATH="/Users/hardy/.minishift/cache/oc/v1.5.0/oc:$PATH"
    # Run this command to configure your shell:
    # eval $(minishift oc-env)

Also we need a check for an existing VM. If no VM exists, we need to print an info message (see _docker-env_). ATM I get:

    $ minishift oc-env
    oc-env command
    export PATH=":$PATH"